### ISSUE TITLE
plugin ompsql ssl connection

### ISSUE THREAD

**Issue by fdefilippo** [rsyslog/rsyslog#4741]
Hi, i did not find anywhere how to configure the ompsql plugin for ssl connections, is it possible?

$ rsyslogd -v
rsyslogd  8.2102.0-5.el8 (aka 2021.02) compiled with:
centos8

thank you.

**Comment by alorbach** [comment]
You mean the "ompgsql" plugin?
There is no SSL support implemented yet as far as I can see

**Comment by ericlenio** [comment]
Yes SSL is supported, as of [this commit](https://github.com/rsyslog/rsyslog/commit/e98e691ad82cde0e0ed68138cfe1f83bfeb638f6)

Side note: on my system, I use `$PrivDropToUser` and this created a conflict: during the postgresql SSL handshake, it always tries to access `$HOME/.postgresql/root.crt`, which results in `permission denied` (because `HOME=/root`, and the unprivileged user cannot read this directory), which makes the handshake fail. I worked around this in my config file by setting a fake value for HOME: `global(environment=["HOME=/no-home"])`.

**Comment by rgerhards** [comment]
Apologies for the very late follow-up. This may no longer be relevant for the original reporter, but I am adding the answer for people who arrive here through search while we work through the backlog. We do not like to close issues just because of age; this is now documented/supported through the PostgreSQL connection string.

Current `ompgsql` supports `conninfo`, and PostgreSQL TLS settings can be passed through the libpq connection string/URI, for example with `sslmode=verify-full` and `sslrootcert=...`. The current documentation includes examples for `conninfo`-based configuration.

So the practical answer is: configure TLS through PostgreSQL/libpq connection parameters, not through a separate rsyslog-specific TLS block for `ompgsql`.

Closing this as answered/implemented. If a current `ompgsql` `conninfo` TLS configuration does not work, please open a new issue with the exact `conninfo` value with secrets redacted, rsyslog version, libpq/PostgreSQL version, and the server-side TLS error.
